modules/gapi/src/streaming/onevpl/utils.cpp

```diff
@@ -0,0 +1,454 @@
+    out << "mfxImplDescription.ImplName: " << idesc.ImplName << std::endl;
+    out << "mfxImplDescription.License: " << idesc.License << std::endl;
+    out << "mfxImplDescription.Keywords: " << idesc.Keywords << std::endl;
+    out << "mfxImplDescription.VendorID: " << idesc.VendorID << std::endl;
+    out << "mfxImplDescription.VendorImplID: " << idesc.VendorImplID << std::endl;
+
+    const mfxAccelerationModeDescription &accel = idesc.AccelerationModeDescription;
+    out << "mfxImplDescription.mfxAccelerationMode.Version: " << static_cast<int>(accel.Version.Major)
+        << "." << static_cast<int>(accel.Version.Minor) << std::endl;
+    for (int mode = 0; mode < accel.NumAccelerationModes; mode++) {
+        out << "mfxImplDescription.mfxAccelerationMode.Mode: " << mfx_accel_mode_to_cstr(accel.Mode[mode]) << std::endl;
+    }
+
+    const mfxDeviceDescription &dev = idesc.Dev;
+    out << "mfxImplDescription.mfxDeviceDescription.Version: " << static_cast<int>(dev.Version.Major)
+        << "." << static_cast<int>(dev.Version.Minor) << std::endl;
+    out << "mfxImplDescription.mfxDeviceDescription.DeviceID: " << dev.DeviceID << std::endl;
+    for (int subdevice = 0; subdevice < dev.NumSubDevices; subdevice++) {
+        out << "mfxImplDescription.mfxDeviceDescription.subdevices.Index: " <<     dev.SubDevices[subdevice].Index << std::endl;
+        out << "mfxImplDescription.mfxDeviceDescription.subdevices.SubDeviceID: " <<  dev.SubDevices[subdevice].SubDeviceID << std::endl;
+    }
+
+    /* mfxDecoderDescription */
+    const mfxDecoderDescription &dec = idesc.Dec;
+    out << "mfxImplDescription.mfxDecoderDescription.Version: " << static_cast<int>(dec.Version.Major)
+        << "." << static_cast<int>(dec.Version.Minor) << std::endl;
+    for (int codec = 0; codec < dec.NumCodecs; codec++) {
+        auto cid = dec.Codecs[codec].CodecID;
+        out << "(*)mfxImplDescription.mfxDecoderDescription.decoder.CodecID: " << cid;//(cid & 0xff) << "." << (cid >> 8 & 0xff) << "." << (cid >> 16 & 0xff) << "." << (cid >> 24 & 0xff)  << std::endl;
+        out << "mfxImplDescription.mfxDecoderDescription.decoder.MaxcodecLevel: " << dec.Codecs[codec].MaxcodecLevel << std::endl;
+        for (int profile = 0; profile < dec.Codecs[codec].NumProfiles; profile++) {
+            out << "mfxImplDescription.mfxDecoderDescription.decoder.Profiles: "
+                << mfx_codec_type_to_cstr(dec.Codecs[codec].CodecID,
+                                          dec.Codecs[codec].Profiles[profile].Profile) << std::endl;
+            for (int memtype = 0; memtype < dec.Codecs[codec].Profiles[profile].NumMemTypes; memtype++) {
+                out << "mfxImplDescription.mfxDecoderDescription.decoder.Profiles.MemDesc.MemHandleType: "
+                    << mfx_resource_type_to_cstr(dec.Codecs[codec].Profiles[profile].MemDesc[memtype].MemHandleType) << std::endl;
```

Comment:
It seems that there are should be intermediate variables for `codec`, `profile` and `memtype`..